helper.php

```diff
@@ -110,8 +110,22 @@ public function getSecret($service) {
      * @param $service
      * @return string
      */
+    public function getUrl($service) {
+        $service = strtolower($service);
+        return $this->getConf($service.'-url');
+    }
+
+    /**
+     * Return the configured Authentication Endpoint URL for the given service
+     *
+     * @param $service
+     * @return string
+     */
     public function getAuthEndpoint($service) {
         $service = strtolower($service);
+        if (strcmp($service, 'Gitlab')) {
+            return ($this->getUrl($service) . '/oauth/authorize');
+        }
```

Comment:
Yeah, I think you are right, there is some logic leaking which should be encapsulated in the adapter.
If `getTokenEndpoint` and `getAuthEndpoint` logic is provided by the adapter in the case of Gitlab, then maybe we should move them into the adapter for Doorkeeper too?